pytorch_lightning/utilities/parsing.py

```diff
@@ -197,72 +197,70 @@ def __repr__(self):
         return out
 
 
-def lightning_hasattr(model, attribute):
-    """ Special hasattr for lightning. Checks for attribute in model namespace,
-        the old hparams namespace/dict, and the datamodule. """
+def lightning_get_all_attr_holders(model, attribute):
+    """ Special attribute finding for lightning.  Gets all of the objects or dicts that holds attribute.
+            Checks for attribute in model namespace, the old hparams namespace/dict, and the datamodule. """
     trainer = getattr(model, 'trainer', None)
 
-    attr = False
+    holders = []
+
     # Check if attribute in model
     if hasattr(model, attribute):
-        attr = True
+        holders.append(model)
+
     # Check if attribute in model.hparams, either namespace or dict
-    elif hasattr(model, 'hparams'):
-        if isinstance(model.hparams, dict):
-            attr = attribute in model.hparams
-        else:
-            attr = hasattr(model.hparams, attribute)
+    if hasattr(model, 'hparams'):
```

Comment:
`hparams` is property so what is the case it would be missing? some old model versions?

Reply:
This function is just collectin `attr` candidates. If there is no `hparams`, one of the previous/next candidates will be used